resources/splunk_app.rb

```diff
@@ -0,0 +1,130 @@
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_app
+#
+# Resource for installing and configuring Splunk apps
+
+class SplunkApp < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers, CernerSplunk::ResourceHelpers
+
+  resource_name :splunk_app
+
+  property :name, String, name_property: true, identity: true
+  property :install_dir, String, required: true, desired_state: false
+  property :package, [:splunk, :universal_forwarder], required: true, desired_state: false
+  property :source_url, String
+  property :version, String
+  property :configs, Proc
+  property :files, Proc
+  property :metadata, Hash, default: {}
+
+  default_action :install
+
+  def install_state
+    unless @install_exists
+      raise 'Attempted to reference service for Splunk installation that does not exist' unless load_installation_state
+      @install_exists = true
+    end
+
+    node.run_state['splunk_ingredient']['installations'][install_dir]
+  end
+
+  def app_path
+    @app_path ||= Pathname.new(splunk_app_path).join(name)
+  end
+
+  def config_scope
+    'local'
+  end
+
+  def parse_meta_access(access)
+    access_read = Array(access[:read] || access['read']).join(', ')
```

Comment:
I suspect it will throw a method not found error. I also noticed that it's not even checking if access is a hash before trying to manipulate it.